qunicorn_core/core/pilotmanager/qmware_pilot.py

```diff
@@ -126,15 +126,20 @@ def _get_job_result(job_id: str, program: QuantumProgramDataclass) -> ResultData
         counts = {element["number"]: element["hits"] for element in results}
         probabilities = {element["number"]: element["probability"] for element in results}
 
-        return ResultDataclass(
-            program=program,
-            result_dict={
-                "counts": Pilot.qubits_decimal_to_hex(counts),
-                "probabilities": Pilot.qubits_decimal_to_hex(probabilities),
-            },
-            result_type=ResultType.COUNTS,
-            meta_data={"data": {"counts": Pilot.qubits_decimal_to_hex(counts)}},  # TODO: save more metadata
-        )
+        return [
+            ResultDataclass(
+                program=program,
+                data=Pilot.qubits_decimal_to_hex(counts),
+                result_type=ResultType.COUNTS,
+                meta={},  # TODO: save more metadata
```

Comment:
Hex format requires some metadata to be present! This comment should at least be a fixme.